backend/server/models.py

```diff
@@ -0,0 +1,48 @@
+import os
+from typing import Literal, Optional
+
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_fireworks import ChatFireworks
+from langchain_openai import ChatOpenAI
+
+SUPPORTED_MODELS = {
```

Comment:
Could we create this dynamically based on which keys are present in the enviornment?